### PR TITLE
solution

### PR DESCRIPTION
**Pull request by Basil-Panasiuk**

### CODE REVIEW

File: src/figuresClasses.ts [hunk 1]
```diff
@@ -1,19 +1,112 @@
...
+  if (partsOfNumber.length > 1) {
+    partsOfNumber[1] = partsOfNumber[1].slice(0, 2);
+  }
+
+  return +partsOfNumber.join('.');
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  shape: Shape;
```
Comment (samson-sim) [inline src/figuresClasses.ts:30]: эти shape можно убрать, и функция округления не очень красивая

**Review by samson-sim — CHANGES_REQUESTED** [review]

File: src/figuresClasses.ts (hunk 1 above)
Comment (Basil-Panasiuk) [reply src/figuresClasses.ts:30]: if delete this line, "shape" property does not exist in the "Triangle"

**Review by samson-sim — APPROVED** [review]

File: src/figuresClasses.ts [hunk 2]
```diff
@@ -1,19 +1,106 @@
...
+        return maxSide !== side
+          ? sum + side
+          : sum;
+      }, 0);
+
+    return maxSide < sumSides;
+  }
 
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
```
Comment (Olexandr6) [inline src/figuresClasses.ts:53]: What is this `s`? Rename it

File: src/figuresClasses.ts [hunk 3]
```diff
@@ -1,19 +1,106 @@
...
+export class Triangle implements Figure {
+  shape: Shape;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    this.shape = Shape.Triangle;
```
Comment (Olexandr6) [inline src/figuresClasses.ts:28]:
```suggestion
  shape = Shape.Triangle

  constructor(
    public color: Color,
    public a: number,
    public b: number,
    public c: number,
  ) {
```

File: src/figuresClasses.ts [hunk 4]
```diff
@@ -1,19 +1,106 @@
...
+  shape: Shape;
+
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    this.shape = Shape.Circle;
+
+    if (this.radius <= 0) {
+      throw new Error('Invalid radius');
```
Comment (Olexandr6) [inline src/figuresClasses.ts:71]:
```suggestion
  shape = Shape.Circle;

  constructor(
    public color: Color,
    public radius: number,
  ) {
    if (this.radius <= 0) {
      throw new Error('Invalid radius');
```

File: src/figuresClasses.ts [hunk 5]
```diff
@@ -1,19 +1,106 @@
...
+    public color: Color,
+    public width: number,
+    public height: number,
+  ) {
+    this.shape = Shape.Rectangle;
+
+    if (this.width <= 0 || this.height <= 0) {
+      throw new Error('Invalid data');
+    }
+  }
```
Comment (Olexandr6) [inline src/figuresClasses.ts:95]:
```suggestion
export class Rectangle implements Figure {
  shape = Shape.Rectangle;

  constructor(
    public color: Color,
    public width: number,
    public height: number,
  ) {
    if (this.width <= 0 || this.height <= 0) {
      throw new Error('Invalid data');
    }
  }
```

**Review by Olexandr6 — CHANGES_REQUESTED** [review]

**Review by Olexandr6 — APPROVED** [review]
Good job 👍

**Pull request closed without merging**